src/geo/projection/globe_util.js

```diff
@@ -274,9 +274,16 @@ export function aabbForTileOnGlobe(tr: Transform, numTiles: number, tileId: Cano
     let arcCenter = new Array(3);
     let closestArcIdx = 0;
 
-    const dx = center[0] - tileCenter[0];
+    let dx = center[0] - tileCenter[0];
     const dy = center[1] - tileCenter[1];
 
+    // Shortest distance might be across the antimeridian
+    if (dx > .5) {
+        dx -= 1;
+    } else if (dx < -.5) {
+        dx += 1;
+    }
```

Comment:
These are slightly different statements compared to native, do you think it would be best not to have diverging code between the two codebases? To make further backporting simpler, refer [here](https://github.com/mapbox/mapbox-gl-native-internal/pull/3450/files#diff-d5ec94bc8ef3b16f2b848cd9120dc63267987828cf517e9edbae8b073e4f43a0R644-R648):

```c++
    auto dx = center[0] - tileCenter[0];
    auto dy = center[1] - tileCenter[1];
    auto xLen = std::abs(dx);

    // Shortest distance might be across the antimeridian
    if (std::abs(dx - 1) < xLen) {
        dx -= 1;
    } else if (std::abs(dx + 1) < xLen) {
        dx += 1;
    }
```

cc @mpulkki-mapbox